### DISCUSSION TITLE
Introducing Anomalib Guru on Gurubase.io

### DISCUSSION THREAD

**Discussion by kursataktas** [open-edge-platform/anomalib#2383] [Show and tell]
Hello team,

I'm the maintainer of [Anteon](https://github.com/getanteon/anteon). We have created Gurubase.io with the mission of building a centralized, open-source tool-focused knowledge base. Essentially, each "guru" is equipped with custom knowledge to answer user questions based on collected data related to that tool.

I wanted to update you that I've manually added the [Anomalib Guru](https://gurubase.io/g/anomalib) to Gurubase. Anomalib Guru uses the data from this repo and data from the [docs](https://anomalib.readthedocs.io/en/latest/) to answer questions by leveraging the LLM.

In this [PR](https://github.com/openvinotoolkit/anomalib/pull/2382), I showcased the "Anomalib Guru", which highlights that Anomalib now has an AI assistant available to help users with their questions. Please let me know your thoughts on this contribution.

Additionally, if you want me to disable Anomalib Guru in Gurubase, just let me know that's totally fine.

**Comment by ashwinvaidya17** [comment]
Can you also add some screenshots (or maybe a small gif/video) here? Would be a great hook for new readers.

**Reply by kursataktas** [reply]
Sure, here are a few screenshots. I asked, "What is Anomalib?", and it provided an answer by referencing the official Anomalib documentation. You can certainly ask more challenging and complex questions as well.

Question asking state
<img width="1457" alt="image" src="https://github.com/user-attachments/assets/806f5c12-bf62-46ea-ad94-1d9ae270ed2d">


Generated Answer 
<img width="1525" alt="image" src="https://github.com/user-attachments/assets/9235e688-c6a4-4db5-9890-97da123fcebf">


The source section lists the webpages used to generate this answer.
<img width="1307" alt="image" src="https://github.com/user-attachments/assets/e3d0a6d0-589f-4f92-8543-7920281a8ba3">

**Comment by kursataktas** [comment]
Hey everyone, have you had a chance to check this out?

**Reply by vmiller987** [reply]
I've gone through it. As someone new to unsupervised training and Anomalib, it is nice having a chat system with access to the Anomalib library. It's certinaly not perfect, but it's been great at letting me ask questions and giving me good answers with examples.

**Comment by kursataktas** [comment]
Hi everyone,

Have you had a chance to check this out?

Additionally, I’d like to update you on the release of the Maintainer Panel feature on Gurubase. With this panel, you can add, remove, or update data sources, as well as change the logo and more. You can find the details about [how to claim a Guru here](https://github.com/getanteon/gurubase?tab=readme-ov-file#how-to-claim-a-guru). 

In the near future, I’m planning to include analytics insights in this panel, such as the number of questions asked, the most frequently asked ones, and more. I’ll be sure to update this thread once it’s available. However, in case I miss it, I highly recommend joining our [Discord channel](https://discord.gg/9CMRSQPqx6) to stay updated.

If you’d prefer that I don’t update this thread anymore, please let me know.

**Reply by vmiller987** [reply]
It's missing certain pieces from the database it seems. It thinks `superimpose_anomaly_map()` is from somewhere else, when it comes from `anomalib.utils.post_processing`

![chrome_3kgyEqYik8](https://github.com/user-attachments/assets/8f9559e6-4bb4-4ad8-9aba-e91cab7077fa)
(The answer to this specific question is that it already does the heatmaps as jet. I incorrectly assumed it was viridis.)

I have tried asking it directly about this function and it's not accurate either.

![chrome_H8BY7K2vLj](https://github.com/user-attachments/assets/64f1824e-a6d9-480e-b724-1efc92e301a6)

**Reply by kursataktas** [reply]
Thanks @vmiller987 for the review. I'll look into this and update this thread.

**Reply by kursataktas** [reply]
Hey @vmiller987 
I found that there is not enough data to generate an answer for this question. To address such situations, a new version of Gurubase was released yesterday. In this version, the system evaluates related data sources before answering a question. If it cannot find sufficient data, it refrains from providing an answer. Additionally, each successfully generated answer includes a "Trust Score," which indicates how confident Gurubase is about the answer. Could you try it again to see how it works?

**Comment by samet-akcay** [comment]
Hi @kursataktas, we were a bit busy with v2 development. Thanks for your patience.

Since the API will change a bit with v2, using Guru would be nice for people to get up to speed with v2, ask questions and so on.
